### PR TITLE
Enhancements

### PR DESCRIPTION
**Pull request by Laerte**
Some enhancements to the module.

- [x] Drop support for Python 3.5 and 3.6
- [x] Use final 3.11 version in CI
- [x] Use latest PyPy version in CI
- [x] Bumped mypy and pylint versions
- [x] Bumped actions version
- [x] Fix Pylint errors after bump
- [x] Simplify envlist

### CODE REVIEW

File: pylintrc [hunk 1]
```diff
@@ -16,3 +16,6 @@ disable=
         missing-function-docstring,
         missing-module-docstring,
         no-self-use,
+        unspecified-encoding,
+
+load-plugins=pylint.extensions.no_self_use
```
Comment (Laerte) [inline pylintrc:21]: https://pylint.pycqa.org/en/latest/whatsnew/2/2.14/summary.html#removed-checkers

File: queuelib/tests/test_queue.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 class DummyQueue:
     def __init__(self) -> None:
-        self.q = list()  # type: list
+        self.q = []  # type: list
```
Comment (elacuesta) [inline queuelib/tests/test_queue.py:23]:
```suggestion
        self.q: list = []
```
This is possible after dropping support for py35

**Review by elacuesta — APPROVED** [review]
Great work, thanks!

File: queuelib/tests/test_queue.py (hunk 2 above)
Comment (Laerte) [reply queuelib/tests/test_queue.py]: Done. 💪🏻

**Review by Gallaecio — APPROVED** [review]

**Comment by Gallaecio** [comment]
Awesome!

**Pull request merged**